tempesta_fw/sock_srv.c

```diff
@@ -521,17 +528,15 @@ tfw_srv_conn_free(TfwSrvConn *srv_conn)
 }
 
 static int
-tfw_sock_srv_add_conns(TfwServer *srv, int conns_n)
+tfw_sock_srv_add_conns(TfwServer *srv)
 {
 	int i;
 	TfwSrvConn *srv_conn;
 
-	for (i = 0; i < conns_n; ++i) {
+	for (i = 0; i < srv->conn_n; ++i) {
 		if (!(srv_conn = tfw_srv_conn_alloc()))
 			return -ENOMEM;
-		tfw_connection_link_peer((TfwConn *)srv_conn,
-					 (TfwPeer *)srv);
-		tfw_sg_add_conn(srv->sg, srv, srv_conn);
```

Comment:
You can't simply add a server or a connection to a scheduler. Scheduler's data is a set of very specific data structures that are built for efficiency. In both cases we need to build a new set of data for a scheduler, and then make a scheduler seamlessly use it via RCU. At least, that's my understanding. Did you have something different in mind?